File: src/models/Booking.java
package models;

import java.time.LocalDateTime;

public class Booking {
    private User user;
    private String room;
    private LocalDateTime startTime;
    private LocalDateTime endTime;

    // Constructor
    public Booking(User user, String room, LocalDateTime startTime, LocalDateTime endTime) {
        this.user = user;
        this.room = room;
        this.startTime = startTime;
        this.endTime = endTime;
    }

    // Getters and Setters
    public User getUser() {
        return user;
    }

    public void setUser(User user) {
        this.user = user;
    }

    public String getRoom() {
        return room;
    }

    public void setRoom(String room) {
        this.room = room;
    }

    public LocalDateTime getStartTime() {
        return startTime;
    }

    public void setStartTime(LocalDateTime startTime) {
        this.startTime = startTime;
    }

    public LocalDateTime getEndTime() {
        return endTime;
    }

    public void setEndTime(LocalDateTime endTime) {
        this.endTime = endTime;
    }

    @Override
    public String toString() {
        return "Booking{" +
                "user=" + user +
                ", room='" + room + '\'' +
                ", startTime=" + startTime +
                ", endTime=" + endTime +
                '}';
    }
}

File: src/app/Booking.js
export class Booking {
  constructor(user, room, startTime, endTime) {
    this.user = user;
    this.room = room;
    this.startTime = new Date(startTime);
    this.endTime = new Date(endTime);
  }

  getDuration() {
    const durationMs = this.endTime - this.startTime;
    const hours = Math.floor(durationMs / (1000 * 60 * 60));
    const minutes = Math.floor((durationMs % (1000 * 60 * 60)) / (1000 * 60));
    return { hours, minutes };
  }
}
